hudi-flink-datasource/hudi-flink/pom.xml

```diff
@@ -89,6 +89,12 @@
             <artifactId>hudi-client-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hudi</groupId>
+            <artifactId>hudi-aws</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
```

Comment:
We still add a dependency for the `hudi-client-common`, if we want to avoid clazz conflicts, a better solution is to create a shaded module for these vendor jars.

Reply:
I was going to take it up after this patch lands. HUDI-4011